routers/api/v1/api.go

```diff
@@ -1522,6 +1522,11 @@ func Routes() *web.Router {
 								Delete(reqToken(), reqAdmin(), repo.UnpinIssue)
 							m.Patch("/{position}", reqToken(), reqAdmin(), repo.MoveIssuePin)
 						})
+						m.Group("/lock", func() {
+							m.Combo("").
+								Put(bind(api.LockIssueOption{}), repo.LockIssue).
```

Comment:
Need `reqAdmin` permission?

Reply:
I'm sorry, I don't know how to manage permissions in Gitea. The function handling the request checks if the user has the right to write issues and pull requests [here](https://github.com/go-gitea/gitea/pull/34165/files#diff-96d4638541354b54d5086b0ac4282a01fa434d351fbba0d69667bc818cedde15R83). But, if you tell me it's best practice to put the permission check in `api.go`, I will change it.

Reply:
Just add `reqAdmin()` here.

Reply:
Done